pkg/agent/monitortool/latency_store.go

```diff
@@ -237,21 +237,38 @@ func (s *LatencyStore) DeleteStaleNodeIPs() {
 	}
 }
 
-// ConvertList converts the latency store to a list of TargetIPLatencyStats.
-func (l *LatencyStore) ConvertList() []stv1aplpha1.TargetIPLatencyStats {
+// ConvertList converts the latency store to a list of PeerNodeLatencyStats.
+func (l *LatencyStore) ConvertList(excludedNodeName string) []statsv1alpha1.PeerNodeLatencyStats {
 	l.mutex.RLock()
 	defer l.mutex.RUnlock()
 
-	entries := make([]stv1aplpha1.TargetIPLatencyStats, 0, len(l.nodeIPLatencyMap))
-	for nodeIP, entry := range l.nodeIPLatencyMap {
-		tempEntry := stv1aplpha1.TargetIPLatencyStats{
-			TargetIP:                   nodeIP,
-			LastSendTime:               metav1.NewTime(entry.LastSendTime),
-			LastRecvTime:               metav1.NewTime(entry.LastRecvTime),
-			LastMeasuredRTTNanoseconds: entry.LastMeasuredRTT.Nanoseconds(),
+	// PeerNodeLatencyStats should be a list of size N-1, where N is the number of Nodes in the cluster.
+	// We will have N-1 entries for each Node, each entry containing the latency stats for the connection,
+	// TargetIPLatencyStats will be a list of size 1 (single-stack case) or 2 (dual-stack case).
+	peerNodeLatencyStatsList := make([]statsv1alpha1.PeerNodeLatencyStats, 0, len(l.nodeIPLatencyMap)-1)
```

Comment:
The capacity should be `len(l.nodeIPLatencyMap)`. The current Node should not be in `l.nodeIPLatencyMap` to begin with, because the event handlers exclude the current Node. For example:
```go
// onNodeAdd is the event handler for adding Node.
func (m *NodeLatencyMonitor) onNodeAdd(obj interface{}) {
	node := obj.(*corev1.Node)
	if m.isCurrentNode(node) {
		return
	}

	m.latencyStore.addNode(node)

	klog.V(4).InfoS("Node added", "Node", klog.KObj(node))
}
```

That being said, you can keep the `excludedNodeName` parameter if you want to, as there is no harm in using some defensive programming here, but I would suggest renaming the parameter from `excludedNodeName` to `currentNodeName`.

Reply:
Thanks, I prefer to keep `currentNodeName` and already update it in the function.

Reply:
You still need to update the slice capacity. It should be `peerNodeLatencyStatsList := make([]statsv1alpha1.PeerNodeLatencyStats, 0, len(l.nodeIPLatencyMap))` and not `peerNodeLatencyStatsList := make([]statsv1alpha1.PeerNodeLatencyStats, 0, len(l.nodeIPLatencyMap)-1)`